src/huggingface_hub/utils/cache_explorer.py

```diff
@@ -0,0 +1,246 @@
+            raise CorruptedCacheException("Snapshots folder corrupted: found a file  !")
+
+        cached_files = []
+        for file_path in revision_path.glob("**/*"):
+            # glob("**/*") iterates over all files and directories -> skip directories
+            if file_path.is_dir():
+                continue
+
+            if not file_path.is_symlink():
+                raise CorruptedCacheException(
+                    "Revision folder corrupted: found a non-symlink file !"
+                )
+
+            blob_path = Path(file_path).resolve()
+            if not blob_path.exists():
+                raise CorruptedCacheException("Blob missing (broken symlink) !")
+
+            if blobs_path not in blob_path.parents:
+                raise CorruptedCacheException(
+                    "Blob symlink points outside of blob directory !"
+                )
+
+            if blob_path not in blob_sizes:
+                blob_sizes[blob_path] = blob_path.stat().st_size
+
+            cached_files.append(
+                CachedFileInfo(
+                    file_name=file_path.name,
+                    file_path=file_path,
+                    size_on_disk=blob_sizes[blob_path],
+                    blob_path=blob_path,
+                )
+            )
+
+        cached_revisions_by_hash[revision_path.name] = CachedRevisionInfo(
+            commit_hash=revision_path.name,
+            size_on_disk=sum(
+                blob_sizes[blob_path]
+                for blob_path in set(file.blob_path for file in cached_files)
+            ),
+            files=sorted(cached_files, key=lambda file: file.file_path),
+        )
+
+    # Scan over `refs` directory
+    if refs_path.exists():
+        # Example of `refs` directory
+        # ── refs
+        #     ├── main
+        #     └── refs
+        #         └── pr
+        #             └── 1
+        if refs_path.is_file():
+            raise CorruptedCacheException("Refs directory cannot be a file !")
+
+        for ref_path in refs_path.glob("**/*"):
+            # glob("**/*") iterates over all files and directories -> skip directories
+            if ref_path.is_dir():
+                continue
+
+            ref_name = str(ref_path.relative_to(refs_path))
+            with ref_path.open() as f:
+                commit_hash = f.read()
+
+            if commit_hash not in cached_revisions_by_hash:
+                raise CorruptedCacheException(
+                    "Reference refers to a missing commit hash !"
+                )
+
+            cached_revision = cached_revisions_by_hash[commit_hash]
+            if cached_revision.refs is None:
+                cached_revision.refs = {ref_name}
+            else:
+                cached_revision.refs.add(ref_name)
+
+    return CachedRepoInfo(
+        repo_id=repo_id,
+        repo_type=repo_type,
+        revisions=sorted(
+            cached_revisions_by_hash.values(), key=lambda revision: revision.commit_hash
+        ),
```

Comment:
kinda same question about the sorting here (any reason you sort?)

Reply:
see above https://github.com/huggingface/huggingface_hub/pull/990#discussion_r954603419